source/production/F0.Minesweeper.Logic/Utilities.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using F0.Minesweeper.Logic.Abstractions;
+
+namespace F0.Minesweeper.Logic
+{
+	static class Utilities
+	{
+		internal static IEnumerable<Location> GetLocationsAreaAroundLocation(IEnumerable<Location> allLocations, Location location, bool exludeGivenLocation)
+		{
+			var returnLocations = new List<Location>();
+			uint x = location.X;
+			uint y = location.Y;
+
+			for (int xi = -1; xi <= 1; xi++)
+			{
+				for (int yi = -1; yi <= 1; yi++)
+				{
+					var locationToLookFor = new Location(
+						(uint)Math.Max(x + xi, 0),
```

Comment:
If we would have the current minefields height and width present here we could additionally use a `Math.Min` for x and y which would remove the necessity of the `SingleOrDefault` and the resulting if check

Reply:
Yes, this is correct. I tried to make this method in a _generic enough_ way to also support irregular shaped fields 🙈

Reply:
But wouldn't irregular shaped fields also have a y location that would have a maximum value? I think we should discuss this one in our next session?